### PR TITLE
Change default for cdn.flush_manually to true

### PR DESCRIPTION
**Pull request by jacobd91**
Addition to resolution for https://github.com/W3EDGE/w3-total-cache/issues/535 to help prevent unnecessary Cloudfront Invalidation charges

### CODE REVIEW

**Review by mavas84 — CHANGES_REQUESTED** [review]
This should be enabled by default if the following CDN type is selected:
Origin Pull: cf2 - Amazon Cloudfront
Origin Push: cf - Amazon CloudFront Over S3
Origin Push: s3 - Amazon Simple Storage Service (S3)

For all other CDN types it should not be enabled by dafault.

The Only Purge CDN Manually checkbox was added specifically because of this problem in the Performance>CDN>Advanced section. Once checked, any changes are not invalidated automatically, so the cache needs to be purged manually as explained above.
This is not, however, enabled by default.
This checkbox should be enabled by default if one of the following CDN types are selected in the Performance>General Settings>CDN section under CDN type:
Origin Pull: Amazon Cloudfront
Origin push: Amazon Cloudfront over S3
Origin Push: Amazon Simple Storage Service (S3)

**Review by mavas84 — APPROVED** [review]

File: inc/options/cdn.php [hunk 1]
```diff
@@ -377,8 +377,11 @@
 
 			<tr>
 				<th colspan="2">
-					<?php $this->checkbox( 'cdn.flush_manually' ); ?>
 					<?php
+					$this->checkbox(
+						'cdn.flush_manually',
+						Cdn_Util::get_flush_manually_default_override( $this->_config->get_string( 'cdn.engine' ) )
```
Comment (cssjoe) [inline inc/options/cdn.php:383]: This argument disables the checkbox.  A user should be able to toggle the setting regardless of the `cdn.engine` selected.
See https://github.com/W3EDGE/w3-total-cache/blob/5a84ed9f83d7a4fe18c4d59453012b1f444eb4d6/Base_Page_Settings.php#L88

File: Cdn_Util.php [hunk 2]
```diff
@@ -230,4 +230,18 @@ static function replace_folder_placeholders_to_uri( $file ) {
...
+	 * Get the override default value for cdn.flush_manually to prevent excessive invalidation charges for S3 CF and CF2.
+	 *
+	 * @since 2.2.6
+	 *
+	 * @param string $cdn_engine CDN engine value.
+	 *
+	 * @return boolean default value override;
+	 */
+	public static function get_flush_manually_default_override( $cdn_engine = null ) {
+		$override_targets = array( 'S3', 'CF', 'CF2' );
```
Comment (cssjoe) [inline Cdn_Util.php:244]: The correct values are: 's3', 'cf', 'cf2'.

**Review by cssjoe — CHANGES_REQUESTED** [review]
If a user selects s3, cf, or cf2, then we should check the `cdn.flush_manually` checkbox and notify the user.
 See the two other comments.

File: inc/options/cdn.php (hunk 1 above)
Comment (jacobd91) [reply inc/options/cdn.php]: Reverted and proper application added to "checkbox" function to override the default boolean for cdn.flush_manually

**Review by cssjoe — APPROVED** [review]

**Pull request merged**